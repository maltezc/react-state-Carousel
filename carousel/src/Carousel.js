import { useState } from "react";
import "./Carousel.css";
import Card from "./Card";


/** Carousel: displays images and arrows to navigate through them
 *
 * Props:
 * - photos: array of {src, caption} objects
 * - title: string describing the collection of images
 *
 * State:
 * - currCardIdx: integer for current card index
 *
 * App --> Carousel --> Card
 */
function Carousel({ photos, title }) {
  const [currCardIdx, setCurrCardIdx] = useState(0);

  const currCard = photos[currCardIdx];
  const total = photos.length;

  //if currCard is equal to 0 then left button should be hidden
  let leftHidden = "";
  if (currCardIdx === 0) {
    leftHidden = "hidden"
  }

  //if currCard is equal to total - 1, then right button should be hidden
  let rightHidden = "";
  if (currCardIdx === total - 1) {
    rightHidden = "hidden"
  }

  //Increments currCardIdx state by 1
  function goForward() {
    setCurrCardIdx(currCardIdx + 1);
  }

  //Decrements currCardIdx state by 1
  function goBackward() {
    setCurrCardIdx(currCardIdx - 1);
  }

  return (
    <div className="Carousel">
      <h1>{title}</h1>
      <div className="Carousel-main">
        <i
          className={`bi bi-arrow-left-circle ${leftHidden}`}
          onClick={goBackward}
        />
        <Card
          caption={currCard.caption}
          src={currCard.src}
          currNum={currCardIdx + 1}
          totalNum={total}
        />
        <i
          className={`bi bi-arrow-right-circle ${rightHidden}`}
          onClick={goForward}
        />
      </div>
    </div>
  );
}

export default Carousel;
