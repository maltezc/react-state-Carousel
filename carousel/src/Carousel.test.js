import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";
import photos from "./photos.js";

test("smoke test", function () {
  render(<Carousel photos=' "src": "/static/media/image1.08e1519e765278b10548.jpg"'
    title="test caption" />)
})

// snapshot test
test("matches snapshot", function () {
  const { container } = render(<Carousel photos=' "src": "/static/media/image1.08e1519e765278b10548.jpg"'
    title="test caption" />)
  expect(container).toMatchSnapshot();
})

it("works when you click on the right arrow", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />

  );

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});


test("demonstrate that leftArrow and rightArrow do the same thing", function () {
  const { debug, container } = render(<Carousel photos={photos} title="testTitle" />)
  debug(container)
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  const leftArrow = container.querySelector("bi bi-arrow-right-circle")
  console.log(container.querySelector("img"))
  fireEvent.click(rightArrow);


  // fireEvent.click(leftArrow);
  // const containerAfterleft = container

  console.log("originalContainer", originalContainer)
  console.log("containerAfterRight", containerAfterRight)

})

