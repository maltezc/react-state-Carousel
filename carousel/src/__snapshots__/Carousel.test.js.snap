// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      test caption
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        />
        <img
          class="Card-image"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          55
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;
